pkg/interface/src/views/apps/links/components/lib/link-preview.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { cite } from '~/logic/lib/util';
+import RemoteContent from "~/views/components/RemoteContent";
+
+import moment from 'moment';
+
+
+const URLparser = new RegExp(
+    /((?:([\w\d\.-]+)\:\/\/?){1}(?:(www)\.?){0,1}(((?:[\w\d-]+\.)*)([\w\d-]+\.[\w\d]+))){1}(?:\:(\d+)){0,1}((\/(?:(?:[^\/\s\?]+\/)*))(?:([^\?\/\s#]+?(?:.[^\?\s]+){0,1}){0,1}(?:\?([^\s#]+)){0,1})){0,1}(?:#([^#\s]+)){0,1}/
```

Comment:
is there any reason for this (giant, unmaintable) regex instead of say 
``` typescript
let hostname = '';
try {
  const parsed = new URL(url);
  hostname = parsed.hostname;
} catch(e) {}

Reply:
[This isn't an introduced change](https://github.com/urbit/urbit/blob/master/pkg/interface/src/views/apps/links/components/lib/link-detail-preview.js#L61-L63), it's on livenet.

If it's the one I originally wrote — I believe we want to parse the hostname for, eg.`gopher://` or `ipfs://` or whatever schema and this doesn't? Or there was some other weird catch? Ancient history really. But out of scope here, better suited for later refactor work

Reply:
Ok sweet, no worries then, I wasn't aware it was already there.